src/boost_histogram/_internal/hist.py

```diff
@@ -337,6 +338,24 @@ def ndim(self) -> int:
         """
         return self._hist.rank()
 
+    def compare(self, hist2: "Histogram") -> bool:
+        if (np.allclose(self.view().shape, hist2.view().shape)):
+            if (np.allclose(self.view(), hist2.view())):
+                if (np.allclose(self.variances(), hist2.variances())):
+                    if (re.search("(?<=storage=).*", str(self.view))[0].split('(')[0] == re.search("(?<=storage=).*", str(hist2.view))[0].split('(')[0]):
+                        if (list(self.axes)==list(hist2.axes)):
+                            return ""
+                        else:
+                            return "The axes [" + "\033[91m" + str(list(self.axes)) + " and " + str(list(hist2.axes)) + "\033[0m" + "] are not equal."
```

Comment:
```suggestion
                            return f"The axes [\033[91m{list(self.axes)} and {list(hist2.axes)}\033[0m] are not equal."
```

F-strings were made for this. :) Also I'd pull out the color formatting into a variable, like `red` and `reset`. Actually, there are lots of concerns with color formatting (needs to be disablable and such), at least rich can auto-format this, so probably leave the color formatting off for the moment.

Reply:
I bet if you do `import rich; rich.print(h.compare(h2))` this would still be colorful. ;)

Reply:
Yes I was going to do something similar (with termcolor), but then though I'd do it without importing a library..